### PR TITLE
Refactor location

### LINKED ISSUE #50: Refactor RequestUserLocation and get its locale language.
Refactor RequestUserLocation and get its locale language.

### PR DESCRIPTION
**Pull request by fesave**
# New Pull Request
### Description

A refactoring of the location task is performed to ask for the location permission when starting the application, in addition to obtaining the user's language based on their location.

### Fixes
#50

### CODE REVIEW

File: data/src/main/java/com/architectcoders/aacboard/data/repository/RegionRepositoryImpl.kt [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
-            locationDataSource.getLastUserRegion()?.lowercase() ?: DEFAULT_USER_REGION
+            locationDataSource.getUserLanguage()?.lowercase() ?: DEFAULT_USER_REGION
         } else {
             DEFAULT_USER_REGION
         }
     }
 
     companion object {
-        private const val DEFAULT_USER_REGION = "us"
+        private const val DEFAULT_USER_REGION = "en"
```
Comment (jmgjmg) [inline data/src/main/java/com/architectcoders/aacboard/data/repository/RegionRepositoryImpl.kt]: Update to DEFAULT_USER_LANGUAGE

File: app/src/main/java/com/architectcoders/aacboard/datasource/local/LocationDataSourceImpl.kt [hunk 2]
```diff
@@ -21,18 +22,22 @@ class LocationDataSourceImpl(private val context: Context) : LocationDataSource
...
+    private fun Location?.toLanguage(): String? {
         val addresses = this?.let {
             geocoder.getFromLocation(latitude, longitude, 1)
         }
-        return addresses?.firstOrNull()?.countryCode
+        val countryCode = addresses?.firstOrNull()?.countryCode
+
+        return Locale.getAvailableLocales().firstOrNull { locale ->
+            locale.country == countryCode
+        }?.language
```
Comment (cjimenezsanchez) [inline app/src/main/java/com/architectcoders/aacboard/datasource/local/LocationDataSourceImpl.kt]: If the user has set a language in the device that doesn't match his location we just return a null language and there for we use `en`?

Maybe we could return the device default locale if there is no matching
```suggestion
        }?.language ?: Locale.getDefault().language
```

File: app/src/main/java/com/architectcoders/aacboard/ui/common/PermissionRequester.kt [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+import android.app.Activity
+import com.karumi.dexter.Dexter
+import com.karumi.dexter.listener.PermissionDeniedResponse
+import com.karumi.dexter.listener.PermissionGrantedResponse
+import com.karumi.dexter.listener.single.BasePermissionListener
+
+class PermissionRequester(private val activity: Activity, private val permission: String) {
+
+    fun request(continuation: (Boolean) -> Unit) {
```
Comment (jmgjmg) [inline app/src/main/java/com/architectcoders/aacboard/ui/common/PermissionRequester.kt]: Does the callback function require a boolean value as parameter? We are not using it when calling request from MainNavHostActivity

File: data/src/main/java/com/architectcoders/aacboard/data/repository/RegionRepositoryImpl.kt (hunk 1 above)
Comment (fesave) [reply data/src/main/java/com/architectcoders/aacboard/data/repository/RegionRepositoryImpl.kt]: Good catch!

File: app/src/main/java/com/architectcoders/aacboard/ui/common/PermissionRequester.kt (hunk 3 above)
Comment (fesave) [reply app/src/main/java/com/architectcoders/aacboard/ui/common/PermissionRequester.kt]: You're right! removed

File: app/src/main/java/com/architectcoders/aacboard/datasource/local/LocationDataSourceImpl.kt (hunk 2 above)
Comment (fesave) [reply app/src/main/java/com/architectcoders/aacboard/datasource/local/LocationDataSourceImpl.kt]: Nice! Changed

**Review by jmgjmg — APPROVED** [review]
Good job!

**Pull request merged**